controller/machine/controller.go

```diff
@@ -300,3 +310,24 @@ func (m *Lifecycle) saveConfig(config *machineconfig.MachineConfig, machineDir s
 
 	return obj, nil
 }
+
+func (m *Lifecycle) isNodeInAppliedSpec(machine *v3.Machine) (bool, error) {
+	cluster, err := m.clusterLister.Get("", machine.Spec.ClusterName)
+	if err != nil {
+		return false, err
+	}
+	if cluster == nil {
+		return false, nil
+	}
+	if cluster.Status.AppliedSpec.RancherKubernetesEngineConfig == nil {
```

Comment:
lets also check that the cluster.deletiontimestamp is nil. If not, return false right away